### PR TITLE
Simplify API and README

### PR DESCRIPTION
**Pull request by bwalter**
- Remove context parameter and JsBridge.start() method
- Implicitly call JsBridge.release() when the bridge is garbage-collected
- Simplified README and examples

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -184,115 +222,109 @@ jsBridge.evaluateNoRetVal("""
...
 
-Note:
-- the JS methods are running asynchronously in the "JS" thread
-- Kotlin: the return value of the methods may only be `Unit`/`void` or a `Deferred`.
-- When the method is suspending and the JS method returns a Promise, the call will suspend until the
-JS promise has been fulfilled (or rejected).
-- Java: calling the method will block the caller thread (if it is not a `void` method) until the result has
-been returned.
+- **XMLHtmlRequest:**<br/>
+Support for network requests using okhttp client internally.
```
Comment (oleksandr-yefremov) [inline README.md:239]:
```suggestion
- **XmlHttpRequest (XHR):**<br/>
Support for XmlHttpRequest network requests using `okhttp` client internally. `okhttp` instance can be injected in `JsBridgeConfig` object. _Note: not all HTTP methods are currently implemented, check the source code for details._ Other network clients are not tested but should work as well (polyfill for [fetch](https://www.npmjs.com/package/whatwg-fetch), [axios](https://github.com/axios/axios#features) uses XHR in browser mode)
```

File: README.md [hunk 2]
```diff
@@ -184,115 +222,109 @@ jsBridge.evaluateNoRetVal("""
...
 
+- **Promise:**<br/>
+Support for ES6 promises (Duktape: via polyfill, QuickJS: built-in). Pending jobs are triggered
+after each evaluation.
 
+- **JS Debugger:**<br/>
+JS debugger support (Duktape-only)
```
Comment (oleksandr-yefremov) [inline README.md:246]:
```suggestion
JS debugger support (Duktape only via VSCode plugin)
```

File: README.md [hunk 3]
```diff
@@ -184,115 +222,109 @@ jsBridge.evaluateNoRetVal("""
...
+val createJsApi: suspend (JsValue, JsonObjectWrapper) -> JsValue
+    = JsValue(jsBridge, "createApi").mapToNativeFunction2()  // JS: globalThis.createApi(nativeApi, config)
+val jsApi: JsApi = createJsApi(nativeApiJsValue, config).mapToNativeObject()
 
+// Call JS API methods
+val msg = jsApi.createMessage()  // (suspending) "Hello Android, the temperature is 37.2 degrees C."
+val sum = jsApi.calcSum(3, 2)  // (suspending) 5
+```
 
 ## License
```
Comment (oleksandr-yefremov) [inline README.md:329]: Copyright (C) 2018-**2020** ProSiebenSat1.Digital GmbH

File: jsbridge/src/androidTest/kotlin/de/prosiebensat1digital/oasisjsbridge/ReadmeTest.kt [hunk 4]
```diff
@@ -139,69 +173,69 @@ class ReadmeTest {
...
-          $nativeApi.setCallback(function(payload) {
-            console.log("Got native event with value=" + payload.value);
-          })
-          $nativeApi.triggerEvent()
-        """.trimIndent())
-    }
+    //interface JsApi: NativeToJsInterface {
+    //    suspend fun calcSum1(a: Int, b: Int): Int
+    //    suspend fun calcSum2(a: Int, b: Int): Int
+    //    fun calcSumDeferred(a: Int, b: Int): Deferred<Int>
```
Comment (oleksandr-yefremov) [inline jsbridge/src/androidTest/kotlin/de/prosiebensat1digital/oasisjsbridge/ReadmeTest.kt:179]: Is this outcommented code still needed?

File: jsbridge/src/main/kotlin/de/prosiebensat1digital/oasisjsbridge/JsValue.kt [hunk 5]
```diff
@@ -307,27 +316,29 @@ internal constructor(
...
+    // Create a native proxy to a JS object
     //
     // Notes:
-    // - the proxy object will be returned even if the JS object is invalid or does not implement
-    // all of the methods of the NativeToJsInterface
-    // - the non-suspend methods of the returned proxy object will be running in the JS thread and
-    // block the caller thread if the return value is not Unit or Deferred
-    inline fun <reified T: NativeToJsInterface> mapToNativeObject(): T {
+    // - the proxy object will be returned even if the JS object is invalid or
+    // does not implement/ all of the methods of the NativeToJsInterface
```
Comment (oleksandr-yefremov) [inline jsbridge/src/main/kotlin/de/prosiebensat1digital/oasisjsbridge/JsValue.kt:330]: That means user has to check whether method/object implements `NativeToJsInterface` correctly on his own?

File: README.md [hunk 6]
```diff
@@ -1,159 +1,199 @@
...
 
+## Installation
 
-Minimal hello world:
-```kotlin
-val jsBridge = JsBridge(application.context)
-jsBridge.start()
-jsBridge.evaluateNoRetVal("console.log('Hello world!');")
-jsBridge.release()
+Add jitpack repository (root gradle):
```
Comment (oleksandr-yefremov) [inline README.md:31]: https://jitpack.io/docs/#badges
We can also add badge on the top of Readme

File: README.md [hunk 7]
```diff
@@ -1,159 +1,199 @@
...
-- `JsValue.evaluateBlocking<T>()`  // blocking
-- `JsValue.evaluateBlocking(Class<*> javaClass)`  // from Java
+See [Example](#example-consuming-a-js-api-from-kotlin).
 
-A JS function can be [mapped to a Kotlin proxy function](#3-calling-js-functions-from-kotlin) via `JsValue.mapToNativeObject()`.
-A JS object can be [mapped to a Java/Kotlin proxy object](#5-using-js-objects-from-javakotlin) via `JsValue.mapToNativeObject()`.
+Note: the native methods are called from the "JS" thread and must properly
+manage the execution context (e.g.: going to the main thread when calling
+UI methods). To avoid blocking the JS thread for asynchronous operations, it
+is possible to return a Deferred.
```
[the remark this answers is not in the document]
Comment (bwalter) [reply README.md:193]: In this case it is JavaScript calling Kotlin/Java methods so there is no suspending method (everything is "blocking"). For asynchronous methods, it needs to explicitly return a Deferred (mapped to JS Promise).

File: jsbridge/src/main/kotlin/de/prosiebensat1digital/oasisjsbridge/JsValue.kt (hunk 5 above)
Comment (bwalter) [reply jsbridge/src/main/kotlin/de/prosiebensat1digital/oasisjsbridge/JsValue.kt:337]: In that case, the registration won't fail if a method does not exist in JS during the registration. The user will still get an IllegalArgumentException exception when trying to call the non-existing method (see `fun testRegisterNativeToJsInterfaceSuspend()` in `JsBridgeText.kt`).

File: README.md (hunk 6 above)
Comment (bwalter) [reply README.md:34]: Great idea! Just added the badges for jitpack and github actions.

**Pull request merged**